src/targets/maven.ts

```diff
@@ -0,0 +1,250 @@
+import { homedir } from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
+import { exec } from 'child_process';
+import { isDryRun } from '../utils/helpers';
+import { withRetry } from '../utils/async';
+import { checkExecutableIsPresent } from '../utils/system';
+
+const GRADLE_PROPERTIES_FILENAME = 'gradle.properties';
+
+const ANDROID_DIST_EXTENSION = '.aar'; // Must include the leading `.`
+const ANDROID_RELEASE_SUBSTR = 'release';
+
+/** Config options for the "maven" target. */
+type MavenTargetConfig = {
+  [key in keyof (typeof RequiredConfig & typeof OptionalConfig)]: string;
+};
+
+enum RequiredConfig {
+  ossrhUsername = 'OSSRH_USERNAME',
+  ossrhPassword = 'OSSRH_PASSWORD',
+  mavenUsername = 'MAVEN_CENTRAL_USERNAME',
+  mavenPassword = 'MAVEN_CENTRAL_PASSWORD',
+}
+
+enum OptionalConfig {
+  distributionsPath = 'MAVEN_DISTRIBUTIONS_PATH',
+  settingsPath = 'MAVEN_SETTINGS_PATH',
+  mavenRepoUrl = 'MAVEN_REPO_URL',
+  mavenRepoId = 'MAVEN_REPO_ID',
+  mavenCliPath = 'MAVEN_CLI_PATH',
+  gradleCliPath = 'GRADLE_CLI_PATH',
+}
+
+// Paths should be relative to the root of the repository
+const DEFAULT_ENV_VARIABLES: Record<string, string> = {
+  MAVEN_DISTRIBUTIONS_PATH: 'distributions/',
+  MAVEN_SETTINGS_PATH: 'scripts/settings.xml',
+  MAVEN_REPO_URL:
+    'https://oss.sonatype.org/service/local/staging/deploy/maven2/',
+  MAVEN_REPO_ID: 'ossrh',
+  MAVEN_CLI_PATH: 'scripts/mvnw.cmd',
+  GRADLE_CLI_PATH: 'gradlew',
+};
+
+/**
+ * Target responsible for uploading files to Maven Central.
+ */
+export class MavenTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'maven';
+  /** Target options */
+  public readonly mavenConfig: MavenTargetConfig;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.mavenConfig = this.getMavenConfig();
+    this.checkRequiredSoftware();
+  }
+
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      // Required env variables
+      ossrhUsername: this.getEnvVarValue(RequiredConfig.ossrhUsername),
+      ossrhPassword: this.getEnvVarValue(RequiredConfig.mavenPassword),
+      // TODO: MAVEN_CENTRAL_* shouldnt be required if the user already has `gradle.properties`
+      mavenUsername: this.getEnvVarValue(RequiredConfig.mavenUsername),
+      mavenPassword: this.getEnvVarValue(RequiredConfig.mavenPassword),
+      // Optional env variables
+      distributionsPath: this.getEnvVarOrDefault(
+        OptionalConfig.distributionsPath
+      ),
+      settingsPath: this.getEnvVarOrDefault(OptionalConfig.settingsPath),
+      mavenRepoUrl: this.getEnvVarOrDefault(OptionalConfig.mavenRepoUrl),
+      mavenRepoId: this.getEnvVarOrDefault(OptionalConfig.mavenRepoId),
+      mavenCliPath: this.getEnvVarOrDefault(OptionalConfig.mavenCliPath),
+      gradleCliPath: this.getEnvVarOrDefault(OptionalConfig.gradleCliPath),
```

Comment:
The repetition here is killing me. This is why I didn't want these to be enums. I'll try to find a better way for this.